core/src/main/java/com/flipkart/poseidon/core/PoseidonResponse.java

```diff
@@ -93,8 +95,41 @@ public String getHeader(String key) {
         return headers;
     }
 
+    public void addMultiValueHeader(String key, String value) {
+        if (value == null) {
+            return;
+        }
+
+        List<String> values = Optional.ofNullable(multiValueHeaders.get(key)).orElseGet(ArrayList::new);
```

Comment:
Do we need to synchronize this, if it is required in case of two different threads adding same key with different values?